pkg/controller/controllercmd/builder.go

```diff
@@ -304,6 +305,15 @@ func (b *ControllerBuilder) Run(ctx context.Context, config *unstructured.Unstru
 		return nil
 	}
 
+	infraStatus, err := clusterstatus.GetClusterInfraStatus(ctx, clientConfig)
+	if err != nil {
+		eventRecorder.Warningf("ClusterInfrastructureStatus", "Failed to get cluster infrastructure status because of %s", err)
+		klog.Warningf("unable to get cluster infrastructure status for leader election (falling back to default): %v", err)
+	} else if infraStatus.ControlPlaneTopology == configv1.SingleReplicaTopologyMode && !b.leaderElection.Disable {
+		snoLeaderElection := leaderelectionconverter.LeaderElectionSNOConfig(*b.leaderElection)
+		b.leaderElection = &snoLeaderElection
```

Comment:
this is stomping the code configured leader election if someone specified their actual values.  Only in `WithleaderElection` do you have sufficient information to know what the user provided so you don't overwrite it.

How you resolve this is up to you/@soltysh, but you cannot unconditionally stomp the user's intent.

Reply:
I altered this a bit, I added a flag approach to keep it straight forward and introduce less things. The idea being if the user provides no timing configs in the  `WithleaderElection` then it's safe to assume we can change those configs, so we fetch the infraStatus to see if we should alter the timings for SNO. @soltysh What do you think about this approach?

Reply:
> How you resolve this is up to you/@soltysh, but you cannot unconditionally stomp the user's intent.

@deads2k that was also my suggestion here: https://github.com/openshift/library-go/pull/1242/files#r754222946

Reply:
With the recent [change](https://github.com/openshift/library-go/pull/1242/files#diff-e78bc214728945a843063f3f0d995c03dd7490a6956486055cc0ce4b965dd806R151 ) @soltysh , I'm using the 0 values in the timings to ensure that if the user has decided to have some customizations in the timings then we won't alter the timings during runtime. That should make sure we don't stomp on the users intent, and if no values were provided, then we can query the cluster and change the values if SNO is detected. That's going to be triggered [here](https://github.com/openshift/library-go/pull/1242/files#diff-e78bc214728945a843063f3f0d995c03dd7490a6956486055cc0ce4b965dd806R317)